tests/test_storage.py

```diff
@@ -0,0 +1,110 @@
+from __future__ import annotations
+
+from pathlib import Path
+from time import sleep
+from typing import TYPE_CHECKING
+from uuid import uuid4
+
+import pytest
+
+if TYPE_CHECKING:
+    from typing import Any, Dict, List
+
+    from supabase import Client, StorageFileAPI, SupabaseStorageClient
+
+
+@pytest.fixture(scope="module")
+def storage_client(supabase: Client) -> SupabaseStorageClient:
+    """Creates the storage client for the whole storage tests run"""
+    return supabase.storage()
+
+
+@pytest.fixture(scope="module", autouse=True)
+def delete_left_buckets(request, storage_client: SupabaseStorageClient):
+    """Ensures no test buckets are left"""
+
+    def finalizer():
+        # Sleep 15 seconds in order to let buckets be deleted before the double-check
+        sleep(15)
+        for bucket in storage_client.list_buckets():
+            if bucket.id.startswith("pytest-"):
+                storage_client.empty_bucket(bucket.id)
+                storage_client.delete_bucket(bucket.id)
+
+    request.addfinalizer(finalizer)
+
+
+@pytest.fixture(scope="module")
+def bucket(storage_client: SupabaseStorageClient) -> str:
+    """Creates a test bucket which will be used in the whole storage tests run and deleted at the end"""
```

Comment:
/nit Was slightly confused by the 8 when I first saw it. Perhaps we could put a comment or a constant somewhere?

Reply:
Sure! We can make a method get_uuid that unifies this. It's a common practice since the uuid type 4 is structured into different parts joined by hyphens, and the first part has a lenght of 8 digits, so it's a way to get a 8 digits long unique identifier that doesn't use hyphens.

Reply:
Okay so this wasn't as I explained in the last message. The hex method returns the whole uuid directly without hypens, what returns the uuid with hyphens is  `str(uuid.uuid4())`. Therefore, the only purpose of getting 8 digits is to make the uuid shorter.